web/src/js/ducks/flows/index.ts

```diff
@@ -0,0 +1,366 @@
+// In benchmarking we found no major difference between Map/Set and objects, so we use objects
+// for redux compatibility.
+export default function flowsReducer(
+    state = defaultState,
+    action: UnknownAction,
+): FlowsState {
+    if (FLOWS_RECEIVE.match(action)) {
+        const { sort } = state;
+        const list = action.payload;
+        const _listIndex = buildIndex(list);
+        const byId = new Map(list.map((f) => [f.id, f]));
+        // No filter information yet, we expect that to come immediately after.
+        const view = list.toSorted(makeSort(sort));
+        const _viewIndex = buildIndex(view);
+        const selected = state.selected
+            .map((flow) => byId.get(flow.id))
+            .filter((f) => f !== undefined);
+        const selectedIds = buildLookup(selected);
+        const highlighted = new Set<string>();
+
+        return {
+            list,
+            _listIndex,
+            byId,
+            view,
+            _viewIndex,
+            sort,
+            selected,
+            selectedIds,
+            highlighted,
+        };
+    } else if (FLOWS_ADD.match(action)) {
+        const { flow, matching_filters } = action.payload;
+        const { sort, selected, selectedIds } = state;
+        let { view, _viewIndex, highlighted } = state;
+        // Update list
+        const _listIndex = new Map(state._listIndex);
+        _listIndex.set(flow.id, state.list.length);
+        const list = [...state.list, flow];
+        const byId = new Map(state.byId);
+        byId.set(flow.id, flow);
+        // Update view
+        if (matching_filters[FilterName.Search] !== false) {
+            ({ view, _viewIndex } = insertViewItem(
+                view,
+                _viewIndex,
+                flow,
+                makeSort(sort),
+            ));
+        }
+        // Update highlight
+        if (matching_filters[FilterName.Highlight] === true) {
+            highlighted = new Set(highlighted);
+            highlighted.add(flow.id);
+        }
+
+        return {
+            list,
+            _listIndex,
+            byId,
+            view,
+            _viewIndex,
+            sort,
+            selected,
+            selectedIds,
+            highlighted,
+        };
+    } else if (FLOWS_UPDATE.match(action)) {
+        const { flow, matching_filters } = action.payload;
+        const { _listIndex, sort, selectedIds } = state;
+        let { view, _viewIndex, selected, highlighted } = state;
+        // Update list
+        const listPos = state._listIndex.get(flow.id)!;
+        const list = [...state.list];
+        list[listPos] = flow;
+        const byId = new Map(state.byId);
+        byId.set(flow.id, flow);
+        // Update view
+        const prevViewPos: number | undefined = _viewIndex.get(flow.id);
+        const hasOldFlow = prevViewPos !== undefined;
+        const hasNewFlow = !(matching_filters[FilterName.Search] === false);
```

Comment:
why not simply `const hasNewFlow = matching_filters[FilterName.Search] !== false` ?